src/content/en/fundamentals/engage-and-retain/push-notifications/subscribing-a-user.md

```diff
@@ -0,0 +1,396 @@
+`userVisibleOnly` key or pass in `false` you'll get the following error:
+
+> Chrome currently only supports the Push API for subscriptions that will result in
+user-visible messages. You can indicate this by calling
+`pushManager.subscribe({userVisibleOnly: true})` instead. See
+[https://goo.gl/yqv4Q4](https://goo.gl/yqv4Q4) for more details.
+
+It's currently looking like blanket silent push will never be implemented in Chrome. Instead,
+spec authors are exploring the notion of a budget API which will be allow web apps a certain
+number of silent push messages based on the usage of a web app.
+
+
+
+### applicationServerKey Option
+
+We briefly mentioned the notion "application server keys" in the previous section. "Application
+server keys" are used by a push service to identify the application subscribing a user and
+ensuring that the same application is messaging that user.
+
+Application server keys are a public and private key pair that are unique to your application.
+The private key should be kept a secret to your application and the public key can be shared
+freely.
+
+The `applicationServerKey` option passed into the `subscribe()` call is the applications public
+key. The browser passes this onto a push service when subscribing the user, meaning the push
+service can tie your applications public key to the users `PushSubscription`.
+
+The diagram below illustrates these steps.
+
+1. You web app is loaded in a browser and you call `subscribe()` passing in your public
+application server key.
+1. The browser then makes a network request to a push service who will generate an endpoint,
+associate this endpoint with the applications public key and return the endpoint to the
+browser.
+1. The browser will add this endpoint to the `PushSubscription` which is returned via the
+`subscribe()` promise.
+
+![Illustration of the public application server key is used in subscribe
+method.](./images/svgs/application-server-key-subscribe.svg)
+
+When you later want to send a push message, you'll need to create an **Authorization** header
+which will contain information signed with your application server's **private key**. When the
+push service
+receives a request to send a push message, it can validate this signed **Authorization** header
+by looking up the public key linked to the endpoint receiving the request. If the signature is
+valid the push service knows that it must have come from the application server with the
+**matching private key**. It's basically a security measure that prevents anyone else sending
+messages to an applications users.
+
+![Illustration of how the private application server key is used when sending a
+message.](./images/svgs/application-server-key-send.svg)
+
+Technically, the `applicationServerKey` is optional. However, the easiest
+implementation on Chrome requires it and other browsers may require it in
+the future. It's optional on Firefox.
```

Comment:
'in Chrome' 'in Firefox'.

'In' is what's usually done. Please check throughout.